rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -1066,4 +1106,56 @@ public Response index() throws URISyntaxException {
         return Response.seeOther(new URI("doc/jaxrsdocs/rm/index.html")).build();
     }
 
+    private void prepareInfraParams(@FormParam("infrastructureType") String infrastructureType,
+            @FormParam("infrastructureParameters") String[] infrastructureParameters,
+            @FormParam("infrastructureFileParameters") String[] infrastructureFileParameters, ResourceManager rm,
+            Object[] infraParams) {
+        /*
+         * we need to merge both infrastructureParameters and infrastructureFileParameters into one
+         * to do so we need the infrastructure parameter order from the RM
+         */
+        for (PluginDescriptor infra : rm.getSupportedNodeSourceInfrastructures()) {
```

Comment:
Can you refactor this code. 

ideal levels of nesting for the "for block" is 1 ;), 3 should be a maximum

Reply:
done